lib/src/main/java/net/ypresto/androidtranscoder/MediaTranscoder.java

```diff
@@ -153,7 +162,7 @@ public void transcodeVideo(final FileDescriptor inFileDescriptor, final String o
         Looper looper = Looper.myLooper();
         if (looper == null) looper = Looper.getMainLooper();
         final Handler handler = new Handler(looper);
-        mExecutor.execute(new Runnable() {
+        mFuture = mExecutor.submit(new Runnable() {
```

Comment:
You cannot cancel other than the latest transcoding session if you keep Future in field.
Future should be returned to caller to give control around cancelation and timeout.